src/stubgen.py

```diff
@@ -62,11 +63,11 @@ class and repeatedly call ``.put()`` to register modules or contents within the
 class StubGen:
     def __init__(
         self,
-        include_docstrings=True,
-        include_private=False,
-        module=None,
-        max_expr_length=50,
-        patterns={},
+        include_docstrings: bool = True,
+        include_private: bool = False,
+        module: Optional[types.ModuleType] = None,
+        max_expr_length: int = 50,
+        patterns: "dict[re.Pattern[str], tuple[list[str], int]]" = {},
```

Comment:
Note that list becomes tuple, as the first item must be list[str] while second item must be int. Using list[Union[list[str], int]] would make type checkers unhappy.

Also, this is complex type. While this works, consider using a class instead of dictionary?